src/connectionManager.ts

```diff
@@ -0,0 +1,74 @@
+export interface Context {
+  id: string;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: string;
+  url: URL;
+  context: Context[];
+}
+
+export class ConnectionManager {
+  private connections: Connection[];
+  private currentConnection?: Connection;
+
+  constructor() {
+    this.connections = [];
+  }
+
+  public getAllConnections(): Connection[] {
+    return this.connections;
+  }
+
+  public getCurrentConnection(): Connection | undefined {
+    return this.currentConnection;
+  }
+
+  public setCurrentConnection(connection: Connection | undefined) {
+    this.currentConnection = connection;
+  }
+
+  public addConnection(connection: Connection): void {
+    this.connections?.push(connection);
+  }
+
+  // TODO
+  public testConnection(connection: Connection): void {}
+
+  // TODO
+  public upsertConnection(connection: Connection): void {}
+
+  // TODO
+  public updateConnection(connection: Connection): void {}
+
+  public deleteConnection(connection: Connection): void {
+    this.connections.splice(
+      this.getAllConnections()
+        .map(function (x) {
+          return x.id;
+        })
+        .indexOf(connection.id),
```

Comment:
If we're only using the connection id, we should just take that as the parameter here, not an entire Connection.

Also, what happens if the connection id is not found? I believe if `splice` is handed a negative index (like `-1`), it will remove starting from the end of the array.